### PR TITLE
Upgrade Python 3.11

### PR DESCRIPTION
**Pull request by dmichaels-harvard**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -9,26 +9,27 @@ packages = [
 ]
 
 [tool.poetry.dependencies]
-# Cannot go to Python 3.8 because cgap-pipeilne is stuck on Python 3.7.
-python = ">=3.7.1,<3.8"
-#dcicutils = "^7.5.1"
-dcicutils = "7.6.0.1b4"
+python = ">=3.7.1,<3.12"
```
Comment (netsettler) [inline pyproject.toml:12]: drop 3.7?

**Review by netsettler — APPROVED** [review]
.github/workflows/main-CI.yml needs to be updated.
python 3.7 support needs to go away.
probably an ubuntu 20.04 ref in there that could become 22.04
otherwise ok.

**Review by willronchetti — COMMENTED** [review]
Needs build update for new Python version (main.yml)

File: pyproject.toml (hunk 1 above)
Comment (dmichaels-harvard) [reply pyproject.toml]: Done

**Pull request merged**